doc/man7/EVP_MD-EXTERNAL-MU-ML-DSA.pod

```diff
@@ -0,0 +1,143 @@
+=pod
+
+=head1 NAME
+
+EVP_MD-EXTERNAL-MU-ML-DSA - The ExternalMu-ML-DSA.Prehash() EVP_MD implementation
+
+=head1 DESCRIPTION
+
+Support for computing ExternalMu-ML-DSA.Prehash() through the B<EVP_MD> API.
+
+Normally the value of C<mu> is calculated internally as part of an ML-DSA
+sign or verify operation. Mu is defined as:
+    mu = SHAKE256(tr || M', 64)
+
+Where B<tr> is the hash of the encoded public key, and, for Pure (ML-DSA):
+    M' = 0x00 || ctx_len || ctx || message
+
+In cases where prehashing the message is required, FIPS 204 allows
+the C<mu> calculation to be done externally and then C<mu> can be passed to
+ML-DSA sign or verify operations.
+
+PreHash (Hash-ML-DSA) is also supported and uses:
+    M' = 0x01 || ctx_len || ctx || OID || HashedMessage
+
+The output C<mu> value can then be supplied as an input to L<EVP_SIGNATURE-ML-DSA(7)>
+using the ML-DSA Signature Parameter B<OSSL_SIGNATURE_PARAM_MU> (i.e. C<mu>).
+This allows larger messages to be hashed (or hidden) before they are passed to
+pure ML-DSA sign or verify operations.
+
+=head2 Identities
+
+This implementation is available with the FIPS provider as well as the
+default provider, and is identified with the name "ExternalMu-ML-DSA".
+
+=head2 Parameters
+
+This implementation supports the following settable L<OSSL_PARAM(3)> entries:
+
+=over 4
+
+=item "pub" (B<OSSL_DIGEST_PARAM_MU_PUB_KEY>) <octet string>
+
+A B<ML-DSA> encoded public key value of size 1312, 1952 or 2592 bytes
+depending on the respective key type of B<ML-DSA-44>, B<ML-DSA-65> or B<ML-DSA-87>.
+This can be retrieved from a L<EVP_PKEY-ML-DSA(7)> key by calling
+EVP_PKEY_get_octet_string_param(key, OSSL_PKEY_PARAM_PUB_KEY, pub, sizeof(pub), &publen)
+This option MUST be set or an error will occur.
+
+=item "context-string" (B<OSSL_DIGEST_PARAM_MU_CONTEXT_STRING>) <octet string>
+
+An optional string of octets with length at most 255. By default it is the empty string.
+
+=item "digest" (B<OSSL_DIGEST_PARAM_MU_DIGEST>) <utf8 string>
+
+An optional parameter related to "Hash-ML-DSA" that is used to fetch a digest
+algorithm associated with an OID. (See the definition of PreHash M' above).
+By default it is not used.
```

Comment:
```suggestion
If no OID is specified, a Pure ML-DSA C<mu> is computed over the full message,
otherwise the input data must be the result of prehashing the message with the corresponding
digest algorithm.
```